src/domain/siwe/siwe.repository.ts

```diff
@@ -0,0 +1,82 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { CacheRouter } from '@/datasources/cache/cache.router';
+import {
+  CacheService,
+  ICacheService,
+} from '@/datasources/cache/cache.service.interface';
+import { IAuthApi } from '@/domain/interfaces/auth-api.interface';
+import { ISiweRepository } from '@/domain/siwe/siwe.repository.interface';
+import { VerifyAuthMessageDto } from '@/routes/auth/entities/verify-auth-message.dto.entity';
+import { Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class SiweRepository implements ISiweRepository {
+  private readonly nonceTtlInSeconds: number;
+
+  constructor(
+    @Inject(IAuthApi)
+    private readonly authApi: IAuthApi,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(CacheService) private readonly cacheService: ICacheService,
+  ) {
+    this.nonceTtlInSeconds = this.configurationService.getOrThrow(
+      'auth.nonceTtlSeconds',
+    );
+  }
+
+  /**
+   * Generates a unique nonce and stores it in cache for later verification.
+   *
+   * @returns nonce - unique string to be signed
+   */
+  async generateNonce(): Promise<{ nonce: string }> {
+    const nonce = this.authApi.generateNonce();
+
+    // Store nonce for reference to verify/prevent replay attacks
+    const cacheDir = CacheRouter.getAuthNonceCacheDir(nonce);
+    await this.cacheService.set(cacheDir, nonce, this.nonceTtlInSeconds);
```

Comment:
This should be part of the functionality of the `IAuthApi`. The `SiweRepository` shouldn't need to know that this is stored in a cache.

Reply:
I moved all nonce interfacing to the `IAuthApi` in e0ee482bfd60773bb9423aac0cb5252b157da667.